src/Polysemy/CanonicalEffect.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE UndecidableInstances       #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE AllowAmbiguousTypes        #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-|
+Module      : PolySemy.CanonicalEffects
+Description : Polysemy Class for "absorbing" mtl-style constraints.
+
+The CanonicalEffect class represents a choice of polysemy
+interpretation of standard mtl effects as well as a vehicle
+for writing similar "constraint absorbers" for other effects
+with mtl-style constraints.  This class is only useful for
+absorbing single effects.
+
+Consider a function of the type
+f :: MonadReader env m => Int -> m a
+
+It can clearly be "handled by a Polysemy Reader effect
```

Comment:
unterminated quote?

Reply:
Sorry! Forgot to finish the description. Will fix.